### PR TITLE
Recipe App | Development to Main

### PR DESCRIPTION
**Pull request by Rysth**
# 🏅 Features Implemented

Check out commits and branches here: https://github.com/Rysth/Rails-Recipe-App/pulls?q=is%3Apr+is%3Aclosed
List of implemented features based on project requirements:

## 📝 General Requirements

- ✅ No linter errors.
- ✅ Utilized GitFlow methodology appropriately.
- ✅ Crafted a comprehensive and professional `README` file.

## 📝 Project Requirements

- ✅ Added CDN for Daisy UI
- ✅ Improved navigation bar.
- ✅ Created the `Login` and `Register` views.
- ✅ Configurated SendGrid to handle email confirmations.
- ✅ Added `authenticate_user!` to validate that the user is logged in.
- ✅ Redirected not authenticated users to `users/sign_in`.
- ✅ Should display a list of recipes created by the logged-in user as in the wireframe.
- ✅ This should lead to recipe details.
- ✅ Created the `Public` view.
- ✅ Configurated the `Public` controller to get only the Recipes which has `public` as `true`.
- ✅ Allowed only the `User` of the recipe to delete them.
- ✅ Created controllers for Food, shopping lists, and recipe foods.
- ✅ Created views related to foods index and new.
- ✅ Created the `Recipe` show view.
- ✅ Allowed only the `User` to add new ingredients.
- ✅ Validated if the `current_user` is the same as the owner of the `recipe` to allow them to modify their recipes
- ✅ Toggled the `status` from a recipe to public or private.
- ✅ Created the `Navigation` partial view.
- ✅ Logged users can see the `My Recipes`  link and `Logout` button.
- ✅ Not Logged users can see only the `Public Recipes` link and `Login` button.
- ✅ Enhanced performance with eager loading.
- ✅ Improved navigation bar.
- ✅ Fix N+1 problems for `Public`, `Recipe`, and `Foods`.
- ✅ Created a shopping list view
- ✅ Added units testing suites for models ´food´, ´recipe´, ´recipe_food´, and ´user´
- ✅ Created integrated testing suites for views, foods#index, public#index, and recipes#index
- ✅ Linters passed

### Devise

- Login
- image

- Sign Up
- image

- Confirmation
- Screenshot_2

 ### View 
- Public Recipes
- image

- My Recipes
- image

- My Recipes | Add Ingredient Form
- image

- Shopping List
- image

- Food List
- image

### N+1 Solutions

- Public
- public_before
- public_after

- Recipe
- recipe_show

- Foods
- foods_before
- foods_after

### Tests
- general_tests

[_John Palacios_](https://www.linkedin.com/in/john-palacios-rysthcraft/) | _RysthCraft_

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -122,7 +122,14 @@ To run the project, execute the following command:
...
 
-- [ ] **Coming soon.**
+- [ ] 📅 Plan weekly or monthly meal schedules
+- [ ] 📤 Share recipes with friends or social media
+- [ ] 📱 Mobile app support for on-the-go access
+- [ ] 🌐 Integration with online grocery services
+- [ ] 🔄 Synchronize data across multiple devices
+- [ ] 🤖 AI-powered recipe recommendations
+- [ ] 🌱 Sustainability and nutrition insights for recipes
+- [ ] 🎨 Customize the app's theme and layout
```
Comment (andersonlebon) [inline README.md:133]: image

- [ ] Kindy add a form to create an new `recipe`, I have checked your routes and it allows the `/recipes/new/` route but the route is giving an error and there is no any view associated to that route, so kindly work on that by adding a button that will link to the form that will be used for creating a new `recipe`

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @Rysth && @Rebelzob 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

almost-there

## Highlights 🍾 

- Nice use of routes and redux 👏  ✔️ 
- Nice UI ✔️ 
- Project is well structured, and the routes ✔️ 

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md (hunk 1 above)
Comment (Rysth) [reply README.md:133]: @andersonlebon Thanks. The button was added in `Recipes#index` and the form was created successfully.

image

image

image

**Review by joel-kalema — APPROVED** [review]
Hi Team

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 


## Highlights
Linters are passing 👍 
The project is running successfully 👍 
PR is well descriptive 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@joel-kalema** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**